src/python/Services/ReqMgr/ReqMgrReader.py

```diff
@@ -6,53 +6,383 @@
 
 import logging
 import os
+import time
+import copy
 from Utils.WebTools import getResponse
 from Utils.ConfigurationHandler import ConfigurationHandler
 
+from typing import List, Optional, Union, Any, Callable
+
 
 class ReqMgrReader(object):
     """
     _ReqMgrReader_
     General API for reading data from ReqMgr
     """
 
-    def __init__(self, url=None, logger=None, **contact):
-
+    def __init__(
+        self, url: str = None, logger: Optional[logging.Logger] = None, **contact
+    ) -> None:
+        # TODO: url is not being used
         try:
             configurationHandler = ConfigurationHandler()
-            self.reqmgrUrl = os.getenv('REQMGR_URL', configurationHandler.get('reqmgr_url'))
+            self.reqmgrUrl = os.getenv(
+                "REQMGR_URL", configurationHandler.get("reqmgr_url")
+            )
+            self.reqmgrEndpoint = {
+                "req": "/reqmgr2/data/request/",
+                "info": "/reqmgr2/data/info/",
+                "agentConfig": "/reqmgr2/data/wmagentconfig/",
+                "splitting": "/reqmgr2/data/splitting/",
+                "couchdb": "/couchdb/reqmgr_workload_cache/",
```

Comment:
Hasan, this `couchdb` is one example that can be replaced by `/reqmgr2/data/request/`